src/app/codegen-interaction-model/BUILD.gn

```diff
@@ -0,0 +1,27 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import("//build_overrides/chip.gni")
+# This source set is TIGHLY coupled with code-generated data models
+# as generally implemented by `src/app/util`
+#
+# Corresponding functions defined in attribute-storace.cpp/attribute-table.cpp must
+# be available at link time for this model to use
+#
+# Use `model.gni` to get access to:
+#   Model.cpp
+#   Model.h
+#
+# The abolve list of files exists to satisfy the "dependency linter"
```

Comment:
Does the dependency linter just look for file names in the .gn file? or is there some other build sorcery going on?

Reply:
These are listed in model.gni though...does the linter just not consider gni files?

Reply:
It does not ... https://github.com/project-chip/connectedhomeip/blob/master/scripts/tools/not_known_to_gn.py#L148

its intent is to ensure we have a nice tree dependency so it explicitly does not try to find if you reference something from some other build file that is at a different tree level and only considers BUILD.gn files and nothing else. So what I have here is pretending "BUILD.gn" references it while the gni file actually contains something that gets them referenced from outside.

Reply:
I could also update the linter or just accept these files as "orphaned" (since there are the same as data_model.gni probably).

What would you find the best as a code reader?

Reply:
Marking resolved for now as "answered" since I was unsure if any code changes would be appropriate. Willing to make any code changes that improve readability though. Please re-open and suggest if you think code changes would benefit readability or maintenance.